Controllers/ImageController.cs

```diff
@@ -0,0 +1,336 @@
+            }
+        }
+
+        private static string ReadStringFromConfigFile(string filePath)
+        {
+            using (var fs = new FileStream(filePath, FileMode.Open, FileAccess.Read))
+            using (var sr = new StreamReader(fs))
+            {
+                return sr.ReadToEnd();
+            }
+        }
+
+        private IEnumerable<string> GetConfig(string platformId)
+        {
+            List<string> config = new List<string>();
+            string webRootPath = _webHostEnvironment.WebRootPath ?? _webHostEnvironment.ContentRootPath;
+            var root = Path.Combine(webRootPath, "App_Data");
+
+            string filePath = Path.Combine(root, platformId + "Images.json");
+            config.Add(ReadStringFromConfigFile(filePath));
+            return config;
+        }
+
+        private IReadOnlyList<Profile> GetProfilesFromPlatforms(IEnumerable<string> platforms)
+        {
+            List<Profile> profiles = null;
+            foreach (var platform in platforms)
+            {
+                // Get the platform and profiles
+                var config = GetConfig(platform);
+                if (config.Count() < 1)
+                {
+                    throw new HttpRequestException(HttpStatusCode.BadRequest.ToString());
+                }
+
+                foreach (var cfg in config)
+                {
+                    if (profiles == null)
+                    {
+                        profiles = JsonConvert.DeserializeObject<List<Profile>>(cfg);
+                    }
+                    else
+                    {
+                        profiles.AddRange(JsonConvert.DeserializeObject<List<Profile>>(cfg));
+                    }
+                }
+            }
+
+            return profiles;
+        }
+
+        private string CreateFilePathFromId(Guid id)
+        {
+            string webRootPath = _webHostEnvironment.WebRootPath ?? _webHostEnvironment.ContentRootPath;
+            var root = Path.Combine(webRootPath, "App_Data");
+      /*      string root = HttpContextHelper.Current.Server.MapPath("~/App_Data");*/
+            string zipFilePath = Path.Combine(root, id + ".zip");
+            return zipFilePath;
+        }
+
+        private static IImageEncoder getEncoderFromType(string type)
+        {
+            if (!string.IsNullOrEmpty(type))
+            {
+                if (new Regex(type).IsMatch("png"))
+                    return new PngEncoder();
+                if (new Regex(type).IsMatch("jpeg") || new Regex(type).IsMatch("jpg"))
+                    return new JpegEncoder();
+                if (new Regex(type).IsMatch("webp"))
+                    return new WebpEncoder();
+                if (new Regex(type).IsMatch("bmp"))
+                    return new BmpEncoder();
+                if (new Regex(type).IsMatch("tiff"))
+                    return new TiffEncoder();
+            }
+
+            return new PngEncoder();
+        }
```

Comment:
Took a bit to figure out what was trying to be done here. `Regex` shouldn't be used like this. This should work instead:

```suggestion
        private static IImageEncoder getEncoderFromType(string type) => type.ToLower() switch {
             "png" => new PngEncoder(),
             "jpg" or "jpeg" => new JpegEncoder(),
             "webp" => new WebpEncoder(),
             "bmp" => new BmpEncoder(),
             "tiff" => new TiffEncoder(),
             _ => new PngEncoder(),
        };
```

https://learn.microsoft.com/en-US/dotnet/csharp/language-reference/operators/switch-expression

Is this controller cached by ASP? I'm not sure how any of the ASP stuff works. If so, you may want to make these static classes or something so you're not newing up new encoders for every time an image process request comes in?

Reply:
Type could be like this: "image/png", "image/tiff" and etc

Reply:
@khmyznikov will they always be "image/png" or can be "png" or "image/png" you can add extra cases like the "jpg" or "jpeg" there, will be more efficient than using regular expression.

Reply:
Fixed